src/zk.py

```diff
@@ -565,9 +565,11 @@ def get_sync_quorum_hosts(self):
             return []
         return [host for host in all_hosts if self._is_host_in_sync_quorum(host)]
 
-    def get_alive_hosts(self, timeout=1, catch_except=True):
+    def get_alive_hosts(self, timeout=1, catch_except=True, all_hosts_timeout=None):
         ha_hosts = self.get_ha_hosts()
         if ha_hosts is None:
             return []
+        if all_hosts_timeout:
+            timeout = max(timeout, all_hosts_timeout // len(ha_hosts))
```

Comment:
How about adding at least a warning, if `all_hosts_timeout // len(ha_hosts)` is less then `timeout`?

Also, you can exceed overall timeout of `election_timeout / 2` if `timeout` is picked here, even if `all_hosts_timeout = election_timeout / 2`.